persistent-postgresql/test/GeographyType.hs

```diff
@@ -0,0 +1,24 @@
+{-# LANGUAGE OverloadedStrings #-}
+
+-- | This type is taken directly from the example given here: https://hackage.haskell.org/package/persistent-2.10.5/docs/Database-Persist-Types.html#t:PersistValue
+module GeographyType where
+
+import PgInit
+import qualified Data.Text as T
+import qualified Data.ByteString as B
+import qualified Data.ByteString.Char8 as B8
+
+data Geo = Geo ByteString deriving (Show, Eq)
```

Comment:
@ldub You said you concluded that `persist-postgres`'s handling of `PersistDbSpecific` was a bug because of differences in how other SQL backends handle `PersistDbSpecific` for this example. Can we move these test fixtures to a more-central location so that we can run this same test example against all the SQL backends.

Reply:
We can't run this test against any other backend because PostGIS is a postgres-specific example.

However, I can port the `NullableGenerated` test from my other PR. Would that be good?

Reply:
So, what examples, specifically, led you to believe `persist-postgres` was in error? It's something along those lines that we would need to automate.